Source/Scene/Model.js

```diff
@@ -3719,7 +3721,8 @@ define([
             }
         }
 
-        var show = this.show && (this.scale !== 0.0);
+        var displayConditionPassed = defined(this.distanceDisplayCondition) ? this.distanceDisplayCondition.isVisible(this, frameState) : true;
+        var show = this.show && (this.scale !== 0.0) && displayConditionPassed;
```

Comment:
Perhaps slightly faster in the reject case (and the same speed in the accept) to do
```javascript
this.show  && displayConditionPassed && (this.scale !== 0.0)
```